### PR TITLE
Added get_gcs_keys helper function

### PR DESCRIPTION
**Pull request by mlarose**
## Summary & Motivation

- Completion of this feature initiated by https://github.com/dagster-io/dagster/pull/13764

### CODE REVIEW

**Comment by mlarose** [comment]
hi @ajohnson5, as the original contributor of this feature, I thought you might be interested in commenting on this PR.

File: python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    if not gcs_session:
+        gcs_session = storage.Client()
+
+    contents = list(
+        gcs_session.list_blobs(
+            bucket_or_name=bucket,
+            delimiter="",
+            page_size=MAX_KEYS,
+            prefix=prefix,
+        )
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:22]: Should we provide the `start_offset` here, similarly to what we do with `StartAfter` in `get_s3_keys`? See [here](https://github.com/dagster-io/dagster/blob/fc2869650894861b3bec1c541eb4e3a87ce167d2/python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py#L24)

File: python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import dagster._check as check
+from google.cloud import storage
+
+MAX_KEYS = 1000
+
+
+def get_gcs_keys(bucket, prefix="", since_key=None, gcs_session=None):
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:7]: Should we use a cursor like for `get_s3_keys`? See [here](https://github.com/dagster-io/dagster/blob/fc2869650894861b3bec1c541eb4e3a87ce167d2/python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py#L15)

This being said, it seems like the cursor in `get_s3_keys` is not updated through `context.update_cursor`(see example [here](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#sensor-optimizations-using-cursors)) - maybe it's a bug? Or maybe a cursor is not required in both case.

**Review by maximearmstrong — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py (hunk 2 above)
Comment (mlarose) [reply python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:7]: hi, those are good questions but there is a reason why there is no cursor or start_offset aka StartAfter. 

1. this isn't a sensor but a tool function to use in sensor so the cursor in the s3 function isn't the sensor cursor, but the pagination cursor for the s3 api.
2. For GCS, using a cursor to paginate ourselves is not warranted. the list_blobs method returns a fancy little object called a [page iterator](https://googleapis.dev/python/google-api-core/latest/page_iterator.html). That utility pretty much encapsulates all the mechanic of paginating results and tracking the cursor. So here, i am pretty much fetching (by pages of `page_size=MAX_KEYS`) all of the objects in memory by applying list() to the iterator. That's not terribly efficient, but there's no avoiding fetching everything in memory as we're sorting the whole array on updated time right after:

https://github.com/dagster-io/dagster/pull/20191/files/92fc41d87b91563731ef22813947a9d49ebbb4af#diff-9c7f3daad0307d199b33c439bf92c51c9072ce4ab328665aec01823e3d65e70aR25 

There is still however the since_key, which the sensor using this utility function would use to keep its own cursor in context. Here's an example implementing I had used for testing:

```
@sensor(job=my_job, required_resource_keys={"google_cloud_storage_client"})
def my_gcs_sensor(context):
    since_key = context.cursor or None
    new_gcs_keys = get_gcs_keys("mlarose-gcs-test", prefix="data", since_key=since_key, gcs_client=context.resources.google_cloud_storage_client)

    if not new_gcs_keys:
        return SkipReason("No new gcs files found for bucket mlarose-gcs-test.")

    for gcs_key in new_gcs_keys:
        yield RunRequest(run_key=gcs_key, run_config={
            "ops": {
                "gcs_files": {
                    "config": {
                        "gcs_key": gcs_key
                    }
                }
            }
        })

    last_key = new_gcs_keys[-1]
    context.update_cursor(last_key)

```

There are examples in the documentation somewhere for the get_s3_key function that are very similar.

Comment (mlarose) [reply python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:7]: perhaps I should rename the file from `sensor` to `sensor_util` and/or a inline documentation with example usage.

Comment (mlarose) [reply python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:7]: What do you suggest?

**Review by maximearmstrong — APPROVED** [review]
We are good to go after adding a usage example! Thanks @mlarose

**Comment by mlarose** [comment]
> We are good to go after adding a usage example! Thanks @mlarose

Added code example as a stacked PR #20326

**Pull request merged**